cpp/CMakeLists.txt

```diff
@@ -17,54 +17,58 @@ function(open3d_show_and_abort_on_warning trgt)
             /wd4267        # conversion from size_t to smaller type (FixedRadiusSearchCUDA, tests)
             /wd4305        # conversion to smaller type in initialization or constructor argument (examples, tests)
             )
+    set(DISABLE_GNU_CLANG_INTEL_WARNINGS
+            -Wno-unused-parameter               # (many places in Open3D code)
+            -Wno-missing-field-initializers     # (visualization/gui/Events.h)
+            )
 
     if (BUILD_CUDA_MODULE)
-        set(CUDA_FLAGS "--Werror cross-execution-space-call,deprecated-declarations")
+        # General NVCC flags
+        set(DISABLE_NVCC_WARNINGS
+                550            # variable was set but never used (Eigen)
+                940            # missing return statement at end of non-void function (Eigen)
+                2809           # ignoring return value from routine declared with "nodiscard" attribute (Eigen)
+                )
+        string(REPLACE ";" "," DISABLE_NVCC_WARNINGS "${DISABLE_NVCC_WARNINGS}")
 
-        if (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS "10.2")
-            if (MSVC)
-                string(APPEND CUDA_FLAGS " -Xcompiler /W4,/WX")
-            else()
-                string(APPEND CUDA_FLAGS " -Xcompiler -Wall,-Werror")
-            endif()
-        else()
+        set(CUDA_FLAGS "--Werror cross-execution-space-call,deprecated-declarations,reorder")
```

Comment:
`reorder` should not be set for all platforms as it breaks builds on Windows.

Reply:
Fixed, thanks.